analyses/snv-callers/scripts/01-calculate_vaf_tmb.R

```diff
@@ -0,0 +1,294 @@
+tmb_file <- file.path(caller_results_dir, paste0(opt$label, "_tmb.tsv"))
+
+# Declare metadata file name for this caller
+metadata_file <- file.path(
+  caller_results_dir,
+  paste0(opt$label, "_metadata_filtered.tsv")
+)
+
+##################### Check for files if overwrite is FALSE ####################
+# If overwrite is set to FALSE, check if these exist before continuing
+if (!opt$overwrite) {
+  # Make a list of the output files
+  output_files <- c(vaf_file, region_annot_file, tmb_file)
+
+  # Find out which of these exist
+  existing_files <- file.exists(output_files)
+
+  # If all files exist; stop
+  if (all(existing_files)) {
+    stop(cat(
+      "Stopping; --overwrite is not being used and all output files already exist: \n",
+      vaf_file, "\n",
+      region_annot_file, "\n",
+      tmb_file
+    ))
+  }
+  # If some files exist, print a warning:
+  if (any(existing_files)) {
+    warning(cat(
+      "Some output files already exist and will not be overwritten unless you use --overwrite: \n",
+      paste0(output_files[which(existing_files)], "\n")
+    ))
+  }
+}
+
+########################### Set up this caller's data ##########################
+# Print progress message
+message(paste("Reading in", opt$maf, "MAF data..."))
+
+# Read in this MAF, skip the version number
+maf_df <- data.table::fread(opt$maf, skip = 1, data.table = FALSE)
+
+# Print progress message
+message(paste("Setting up", opt$label, "metadata..."))
+
+# Isolate metadata to only the samples that are in the datasets
+metadata <- readr::read_tsv(opt$metadata) %>%
+  dplyr::filter(Kids_First_Biospecimen_ID %in% maf_df$Tumor_Sample_Barcode) %>%
+  dplyr::distinct(Kids_First_Biospecimen_ID, .keep_all = TRUE) %>%
+  dplyr::arrange() %>%
+  dplyr::rename(Tumor_Sample_Barcode = Kids_First_Biospecimen_ID) %>%
+  readr::write_tsv(metadata_file)
+
+# Print out completion message
+message(paste("Filtered metadata file saved to: \n", metadata_file))
+
+# Make sure that we have metadata for all these samples.
+if (!all(unique(maf_df$Tumor_Sample_Barcode) %in% metadata$Tumor_Sample_Barcode)) {
+  stop("There are samples in this MAF file that are not in the metadata.")
+}
+
+################## Calculate VAF and set up other variables ####################
+# If the file doesn't exist or the overwrite option is being used, run this.
+if (any(c(!file.exists(vaf_file), opt$overwrite))) {
+
+  # Print out warning if this file is going to be overwritten
+  if (all(c(opt$overwrite, file.exists(vaf_file)))) {
+    warning("Overwriting existing VAF file.")
+  }
+  # Print out progress message
+  message(paste("Calculating VAF for", opt$label, "MAF data..."))
+
+  # Use the premade function to calculate VAF this will also merge the metadata
+  vaf_df <- set_up_maf(maf_df, metadata) %>%
+    readr::write_tsv(vaf_file)
+
+  # Print out completion message
+  message(paste("VAF calculations saved to: \n", vaf_file))
+} else {
```

Comment:
Personally, I like to put the simpler case first, which in this case would be 

```r
if (file.exists && !overwrite){
  # skip and warn
}else{
  # do the thing which takes a lot of lines
}
```
but that is a style thing. I just like to keep the if and else as close together as possible.

Reply:
I like that idea. Will do.